### PR TITLE
updated SDK to 2023-06-01

### PR DESCRIPTION
**Pull request by JeetendraJoshi-MSFT**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by ArcturusZhang** [comment]
@JeetendraJoshi-MSFT I left a few comments in the corresponding APIview since we would like a stable version of the SDK, please take a look and resolve them: https://apiview.dev/Assemblies/Review/5bcab98a5e4f468eb48f99bfe6fbba8d

Here is some instructions about how to resolve those naming comments:
1. you need to add some configurations following the guidance of the generator here: https://github.com/Azure/autorest.csharp/blob/feature/v3/docs/mgmt/polishing.md#rename-mapping
2. regenerate the code (run command `dotnet build /t:GenerateCode`) and API (run the `eng/scripts/Export-API.ps1 mobilenetwork` script) and samples (run command `dotnet build /t:GenerateTests`)

I suggest we do this in steps, ie resolve a few comments and regenerate, instead of resolving a lot of comment all at once and then regenerate.
Special note about the `Collection, Resource, Data`: there are a few comments for the types ending with the word `Data`, when you change its name, the corresponding `Collection` and `Resource` types will change their name as well, this is by design.

**Comment by andrew-lee-metaswitch** [comment]
@ArcturusZhang:

It seems like you want us to rename all of our classes to be prepended with MobileNetwork (e.g. change DiagnosticsPackageData to MobileNetworkDiagnosticsPackageData say).

I don't understand why we want to do this - all  of these classes are in the Azure.ResourceManager.MobileNetwork namespace so it feels unambiguous what they are referring to.

Is there precedence for this? Is this policy for the dotnet SDK ? if this is policy- why isn't it built into the auto-code-generator to name classes this way (i.e. `dotnet build /t:GenerateCode` could create models/ classes with these names)

My steer here is that as an RP team I want the creation of all the SDKs to be as frictionless and as automated as possible. Happy to talk over teams (andrewlee and jjoshi) if that's easier.

**Comment by ArcturusZhang** [comment]
@andrew-lee-metaswitch all these comments come from the arch board's general comments for mgmt dotnet SDKs
DotNet SDKs should not have same name types across all libraries - this is the main reason we are commenting about "prepend the models with your RP names if the name is not so unique"

The generator has taken over some requirement that must do in its implementations, but it cannot know if the name of the model is general or not therefore this has to be a manual work after the generation

File: sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/MobileNetworkCoreNetworkType.cs [hunk 1]
```diff
@@ -7,12 +7,14 @@
...
 namespace Azure.ResourceManager.MobileNetwork.Models
 {
-    /// <summary> The core network technology generation (5G core or EPC / 4G core). </summary>
-    public enum CoreNetworkType
+    /// <summary> The core network technology generation (5G core, EPC / 4G core or EPC / 4G + 5G core). </summary>
+    public enum MobileNetworkCoreNetworkType
     {
         /// <summary> 5G core. </summary>
-        FiveGC,
+        FiveGC = "5GC",
```
Comment (ArcturusZhang) [inline sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/MobileNetworkCoreNetworkType.cs:14]: this does not seem correct. As far as I know our generator will not do this.
Could you run the generate command again? This should be fixed by the generator.

File: sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/autorest.md [hunk 2]
```diff
@@ -50,13 +50,64 @@ rename-rules:
...
+  SliceData: MobileNetworkSliceData
+  SiteData: MobileNetworkSiteData
+  SimPolicyData: MobileNetworkSimPolicyData
+  SimGroupData: MobileNetworkSimGroupData
+  SimData: MobileNetworkSimData
+  ServiceData: MobileNetworkServiceData
+  PacketCaptureData: MobileNetworkPacketCaptureData
+  DiagnosticsPackageData: MobileNetworkDiagnosticsPackageData
+  DataNetworkData: MobileDataNetworkData
+  AttachedDataNetworkData: MobileAttachedDataNetworkData
```
Comment (ArcturusZhang) [inline sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/autorest.md:109]: These are not taking effect. Could you remove the Data suffix here? Both in key and value

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**